### ISSUE TITLE
Defining a callback to write hessians of train observations to a csv file

### ISSUE THREAD

**Issue by jaguerrerod** [dmlc/xgboost#10144]
I would like to inspect the distribution of hessians during training to understand the impact of the min_child_weight parameter on the split process. Is there a way to define a callback for this using the R interface?

**Comment by trivialfis** [comment]
I think you need to define a custom objective for that.

**Comment by jaguerrerod** [comment]
That's the way when it's possible to define the gradient and hessian within a function, but my issue is with the use of rank:pairwise, where I cannot calculate them within a customizable objective function.
I have no idea about the ranges in which gradients and hessians fluctuate, and therefore, I'll have no choice but to try the regularization parameters alpha, lambda, and min_child_weight with a logarithmic scale to efficiently cover a wide range.
